modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/PreferenceAPIIntegrationUITestCase.java

```diff
@@ -0,0 +1,249 @@
+import org.wso2.identity.integration.test.oauth2.OAuth2ServiceAbstractIntegrationTest;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.io.IOException;
+
+import static org.wso2.identity.integration.test.utils.CommonConstants.DEFAULT_TOMCAT_PORT;
+
+/**
+ * Tests Preference API integration with Login & Recovery Portals.
+ */
+public class PreferenceAPIIntegrationUITestCase extends OAuth2ServiceAbstractIntegrationTest {
+
+    private static final String ENABLE_SELF_REGISTRATION_PROP_KEY = "SelfRegistration.Enable";
+    private static final String ENABLE_USERNAME_RECOVERY_PROP_KEY = "Recovery.Notification.Username.Enable";
+    private static final String ENABLE_PASSWORD_QS_RECOVERY_PROP_KEY = "Recovery.Question.Password.Enable";
+    private static final String ENABLE_PASSWORD_NOTIFICATION_RECOVERY_PROP_KEY =
+            "Recovery.Notification.Password.Enable";
+    private static final String CALLBACK_URL = "https://localhost/callback";
+    private static final String RECOVERY_USERNAME_CONTENT = "id=\"usernameRecoverLink\"";
+    private static final String RECOVERY_PASSWORD_CONTENT = "id=\"passwordRecoverLink\"";
+    private static final String RECOVERY_ENDPOINT_QS_CONTENT = "name=\"recoveryOption\" value=\"SECURITY_QUESTIONS\"";
+    private static final String RECOVERY_ENDPOINT_NOTIFICATION_CONTENT = "name=\"recoveryOption\" value=\"EMAIL\"";
+    private static final String CREATE_ACCOUNT_CONTENT = "id=\"registerLink\"";
+    private static final String RECOVERY_ENDPOINT_URL = "/accountrecoveryendpoint/recoveraccountrouter.do";
+
+    private IdentityProvider superTenantResidentIDP;
+    private IdentityProviderMgtServiceClient superTenantIDPMgtClient;
+    private String recoveryEndpoint;
+    private final String activeTenant;
+    private final String OIDC_APP_NAME = "playground23";
+    private String oidcAppClientId = "";
+
+    @DataProvider(name = "configProvider")
+    public static Object[][] configProvider() {
+
+        return new Object[][]{{TestUserMode.SUPER_TENANT_ADMIN}, {TestUserMode.TENANT_ADMIN}};
+    }
+
+    @Factory(dataProvider = "configProvider")
+    public PreferenceAPIIntegrationUITestCase(TestUserMode userMode) throws Exception {
+
+        super.init(userMode);
+        AutomationContext context = new AutomationContext("IDENTITY", userMode);
+        this.activeTenant = context.getContextTenant().getDomain();
+
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        superTenantIDPMgtClient = new IdentityProviderMgtServiceClient(sessionCookie, backendURL);
+        superTenantResidentIDP = superTenantIDPMgtClient.getResidentIdP();
+        adminClient = new OauthAdminClient(backendURL, sessionCookie);
+        recoveryEndpoint = "https://localhost:9853/t/" + activeTenant + RECOVERY_ENDPOINT_URL;
```

Comment:
The host and port shouldn't be hardcoded.  You may use backendServerURL variable or UrlGenerationUtil.getBackendURL util